compute/cker/include/cker/Shape.h

```diff
@@ -136,6 +136,21 @@ class Shape
     std::memcpy(dst_dims, dims_data, dimensions_count * sizeof(int32_t));
   }
 
+  inline void ReplaceWith(const Shape &other)
+  {
+    ReplaceWith(other.DimensionsCount(), other.DimsData());
+  }
+
+  inline void ReplaceWith(Shape &&other)
+  {
+    Resize(0);
+    std::swap(_size, other._size);
+    if (_size <= kMaxSmallSize)
+      std::copy(other._dims, other._dims + kMaxSmallSize, _dims);
+    else
+      _dims_pointer = other._dims_pointer;
```

Comment:
remove else and replace  this line with 
```suggestion
    std::swap(_dims_pointer, other._dims_pointer);
```
current implementation could lead to:
- double free in case `self._size > kMaxSmallSIze` and `other._size > kMaxSmallSize` (sizes at call time) or segfault on accessing this array
- memory leak in case `self._size > kMaxSmallSIze` and `other._size <= kMaxSmallSize` (sizes at call time)

Reply:
@binarman for https://github.com/Samsung/ONE/pull/4611#discussion_r505117981
I agree now. :-)
Maybe we will find ways to reduce memory when we started to get strong request to reduce memory later. :-)

Reply:
@binarman self._size is always 0 after Resize(0) in the beginning, all memory is freed, so other._size becomes 0 after the size swap.

Reply:
By the way, I don't know the reason why operator= is disabled for the Shape, so I've made new `ReplaceWith` signatures.

Reply:
> this->_size is always 0 after Resize(0)

Then it is ok, thank you.

> By the way, I don't know the reason why operator= is disabled

I think to avoid unintentional memory copies.
Though it was long ago, maybe @hseok-oh can remember the reason.

Reply:
>By the way, I don't know the reason why operator= is disabled

`Shape` class comes from tensorflow lite's `RuntimeShep` class. IMO, it is disabled for blocking unexpected copy.